pkgs/by-name/nc/nchat/package.nix

```diff
@@ -24,16 +24,56 @@ stdenv.mkDerivation rec {
     hash = "sha256-NrAU47GA7ZASJ7vCo1S8nyGBpfsZn4EBBqx2c4HKx7k=";
   };
 
+  libcgowm = buildGoModule {
+    pname = "nchat-wmchat-libcgowm";
+    inherit version;
+
+    inherit src;
+    sourceRoot = "${src.name}/lib/wmchat/go";
+    vendorHash = "sha256-EdbOO5cCDT1CcPlCBgMoPDg65FcoOYvBwZa4bz0hfGE=";
+
+    buildPhase = ''
+      mkdir -p $out/
+      go build -o $out/ -buildmode=c-archive
+      mv $out/go.a $out/libcgowm.a
+      ln -s $out/libcgowm.a $out/libref-cgowm.a
+      mv $out/go.h $out/libcgowm.h
+    '';
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "nchat";
+  inherit version src;
+
+  nl = "\n";
   postPatch = ''
     substituteInPlace lib/tgchat/ext/td/CMakeLists.txt \
       --replace "get_git_head_revision" "#get_git_head_revision"
     substituteInPlace lib/tgchat/CMakeLists.txt \
       --replace-fail "list(APPEND OPENSSL_ROOT_DIR" "#list(APPEND OPENSSL_ROOT_DIR"
+
+    # specific mangling to handle whatsapp go module:
+
+    substituteInPlace CMakeLists.txt \
+      --replace "if(HAS_WHATSAPP AND (NOT GO_VERSION VERSION_GREATER_EQUAL GO_VERSION_MIN))" \
+      "if(FALSE AND (NOT GO_VERSION VERSION_GREATER_EQUAL GO_VERSION_MIN))"
+
+    substituteInPlace lib/wmchat/CMakeLists.txt \
+      --replace-fail "add_subdirectory(go)" \
+    "set(GO_LIBRARIES ${libcgowm}/libcgowm.a)${nl}target_include_directories(wmchat PRIVATE ${libcgowm})"
+
+    substituteInPlace lib/wmchat/CMakeLists.txt \
+      --replace-fail "target_link_libraries(wmchat PUBLIC ref-cgowm ncutil \''${GO_LIBRARIES})" \
+      "target_link_libraries(wmchat PUBLIC ${libcgowm}/libcgowm.a ncutil \''${GO_LIBRARIES})"
+
+    substituteInPlace lib/wmchat/CMakeLists.txt \
+      --replace-fail "add_dependencies(wmchat ref-cgowm)" "#add_dependencies(wmchat ref-cgowm)"
```

Comment:
What about using a patch instead of substitutes? It's easier to maintain.

Reply:
Yeah, I wasn't sure what was preferred, and tbh I'm sufficiently nix-inexperienced that I was c&p from places. I can definitely have a go at reworking it, but can't promise a timescale :(

Reply:
OK, I can do it myself on next update.